### ISSUE TITLE
subject should support emit value by mut ref

### ISSUE THREAD

**Issue by M-Adoo** [rxRust/rxRust#45]
`Observer` emit item by value, for `Subject` we bound the item with `Copy`, that because subject is a multicast stream, need pass item to multi `subscriber`.  So we can emit item `T: Copy` or `&T`, but can't emit `&mut T`.

```rust
// these work
Subject::local().next(1);
Subject::local().next(&1);

// emit `&mut T` will not complied. 
Subject::local().next(&mut 1);
```

But accept an item and change it, it's a useful and normal scene: just subscribe a stream, do some thing for the source emitter (maybe need change its state) when event emit.

```rust
subject.subscribe(|obj: &mut T| {
   obj.change_to_next_state();
   // or
   obj.x = x;
})
```

In logic, this should work, for now `Observable` can emit `&mut T`,  `Subject` can't because `Copy` bound.  But subject only pass the item to multi subscribers, never alias the item, what it really do  is just like : 

```rust
fn next<Item>(&mut self,  value &mut Item) {
   self.observers.for_each(|subscriber |{
       subscriber.next(value);
  })
}
```

But I can't find a way to write an specialization implement `Observer<&mut Item, _>` for `Subject`. 

Should tracking https://github.com/rust-lang/rust/issues/31844

**Comment by M-Adoo** [comment]
#46

**Comment by M-Adoo** [comment]
fixed by #49 and #52
